Xamarin.Forms.Core/Effect.cs

```diff
@@ -20,11 +20,10 @@ internal Effect()
 
 		public static Effect Resolve(string name)
 		{
-			Type effectType;
 			Effect result = null;
-			if (Internals.Registrar.Effects.TryGetValue(name, out effectType))
+			if (Internals.Registrar.Effects.TryGetValue(name, out Type effectType))
 			{
-				result = (Effect)Activator.CreateInstance(effectType);
+				result = DependencyResolver.ForceResolve(effectType) as Effect;
```

Comment:
Suggest considering cast instead of `as`; The result should either be of `effectType` or null, no?

Reply:
Yep. Done.